test/integration/render_task_test.rb

```diff
@@ -0,0 +1,191 @@
+require 'kubernetes-deploy/render_task'
+
+class RenderTaskTest < KubernetesDeploy::TestCase
+  include FixtureDeployHelper
+
+  def test_render_task
+    render = build_render_task(fixture_path('hello-cloud'))
+    fixture = 'configmap-data.yml'
+
+    assert_render_success(render.run([fixture], test_output_stream))
+
+    stdout_assertion do |output|
+      assert_equal output, <<~RENDERED
+        ---
+        apiVersion: v1
+        kind: ConfigMap
+        metadata:
+          name: hello-cloud-configmap-data
+          labels:
+            name: hello-cloud-configmap-data
+            app: hello-cloud
+        data:
+          datapoint1: value1
+          datapoint2: value2
+      RENDERED
+    end
+
+    assert_render_success(render.run([fixture, fixture], test_output_stream))
+  end
+
+  def test_render_task_with_partials_and_bindings
+    render = build_render_task(fixture_path('test-partials'), 'supports_partials': 'yep')
+    fixture = 'deployment.yaml.erb'
+
+    assert_render_success(render.run([fixture], test_output_stream))
+    stdout_assertion do |output|
+      assert_equal output, <<~RENDERED
+        ---
+        apiVersion: extensions/v1beta1
+        kind: Deployment
+        metadata:
+          name: web
+        spec:
+          replicas: 1
+          template:
+            metadata:
+              labels:
+                name: web
+                app: test-partials
+            spec:
+              containers:
+              - name: sleepy-guy
+                image: busybox
+                imagePullPolicy: IfNotPresent
+                command:
+                - sleep
+                - '8000'
+        ---
+        apiVersion: v1
+        kind: Pod
+        metadata:
+          name: pod1
+        spec:
+          restartPolicy: Never
+          activeDeadlineSeconds: 60
+          containers:
+          - name: pod1
+            image: busybox
+            args:
+            - echo
+            - log from pod1
+        ---
+        apiVersion: v1
+        kind: ConfigMap
+        metadata:
+          name: config-for-pod1
+        data:
+          supports_partials: yep
+        ---
+        apiVersion: v1
+        kind: ConfigMap
+        metadata:
+          name: independent-configmap
+        data:
+          value: renderer test
+        ---
+        apiVersion: v1
+        kind: Pod
+        metadata:
+          name: pod2
+        spec:
+          restartPolicy: Never
+          activeDeadlineSeconds: 60
+          containers:
+          - name: pod2
+            image: busybox
+            args:
+            - echo
+            - log from pod2
+        ---
+        apiVersion: v1
+        kind: ConfigMap
+        metadata:
+          name: config-for-pod2
+        data:
+          supports_partials: yep
+      RENDERED
+    end
+  end
+
+  def test_render_task_rendering_all_files
+    render = build_render_task(fixture_path('hello-cloud'))
+
+    assert render.run([], test_output_stream)
```

Comment:
this looks a little weird--I think I'd expect it to render nothing. Maybe this should be `render.run(test_output_stream)`.